spec.html

```diff
@@ -30460,21 +30438,7 @@ <h1>The %StringIteratorPrototype% Object</h1>
         <emu-clause id="sec-%stringiteratorprototype%.next">
           <h1>%StringIteratorPrototype%.next ( )</h1>
           <emu-alg>
-            1. Let _O_ be the *this* value.
-            1. If Type(_O_) is not Object, throw a *TypeError* exception.
-            1. If _O_ does not have all of the internal slots of a String Iterator Instance (<emu-xref href="#sec-properties-of-string-iterator-instances"></emu-xref>), throw a *TypeError* exception.
-            1. Let _S_ be _O_.[[IteratedString]].
-            1. If _S_ is *undefined*, return CreateIterResultObject(*undefined*, *true*).
-            1. Let _position_ be _O_.[[StringNextIndex]].
-            1. Let _len_ be the length of _S_.
-            1. If _position_ &ge; _len_, then
-              1. Set _O_.[[IteratedString]] to *undefined*.
-              1. Return CreateIterResultObject(*undefined*, *true*).
-            1. Let _cp_ be ! CodePointAt(_S_, _position_).
-            1. Let _nextIndex_ be _position_ + _cp_.[[CodeUnitCount]].
-            1. Let _resultString_ be the substring of _S_ from _position_ to _nextIndex_.
-            1. Set _O_.[[StringNextIndex]] to _nextIndex_.
-            1. Return CreateIterResultObject(_resultString_, *false*).
+            1. Return ? GeneratorResume(*this* value, *undefined*, ~String Iterator~).
```

Comment:
I am also not thrilled about having every distinct kind of generator introoduce a spec value along the lines of `~String Iterator~`. I'd prefer that this be, for example, a String which is  
 the name of an intrinsic, like `"%StringIteratorPrototype%"`; that matches what we do for TypedArrays, as in [AllocateTypedArray](https://ci.tc39.es/preview/tc39/ecma262/sha/153e22e51cd700b414682b2652956c0c3593612f/#sec-allocatetypedarray).

I see @ljharb originally asked for this to be changed from a String to a spec enum. @ljharb, thoughts?

Reply:
👀 I'd like to see a clear idea about what name it should be.

Reply:
Sure, I'd be fine with using an intrinsic name. I just didn't want a new kind of magic string.

Reply:
Hmm, that's not a _new_ magic string, it's reflecting the intrinsic value of @@toStringTag (when I wrote them).

So you mean I should change it from `~String Iterator~` to `"%StringIteratorPrototype%"`(a runtime value)?